src/plugins/new-ui/public/webcomponents/power.html

```diff
@@ -216,28 +169,58 @@
           
           element.style.transform=("translate3d(" + config.X + "px," + val + "px,0px )");
         },
-        setVoltage: function (voltage) {
+
```

Comment:
Added a function that calculates a color from a percentage, rather than having hard coded color steps and special logic for calculating it.